File: 2501. Longest Square Streak in an Array/Solution.java
class Solution {
    public int longestSquareStreak(int[] nums) {
        int currentStreak;
        int longestStreak = 0;
        var unique = new HashSet<Integer>();
        Arrays.sort(nums);
        for(int n :  nums){
            unique.add(n);
        }
        for (int n : nums) {
            long current = n;
            currentStreak = 0;
            while(unique.contains( (int) current)){
                currentStreak ++;
                if (current * current > 1e5) break;
                current = current * current;
            }
            longestStreak = Math.max(longestStreak, currentStreak);
        }
        return longestStreak >= 2 ? longestStreak : -1;
    }
}

File: 2501. Longest Square Streak in an Array/Solution.py
class Solution:
    def longestSquareStreak(self, nums: List[int]) -> int:
        longestStreak = 0
        unique = set(nums)
        for n in nums:
            currentStreak = 0
            current =  n
            while current in unique:
                currentStreak += 1
                
                if current * current > 1e5:
                    break
                current *= current
            longestStreak = max(longestStreak, currentStreak)
        return longestStreak  if longestStreak >= 2 else -1
